Sources/TuistGenerator/Mappers/ExplicitDependencyTargetMapper.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+import TuistCore
+import TuistGraph
+import TuistSupport
+
+/// A target mapper that updates
+public struct ExplicitDependencyTargetMapper: TargetMapping {
+    public init() {}
+    
+    public func map(target: Target) throws -> (Target, [SideEffectDescriptor]) {
+        let movedProductNames = target.dependencies.compactMap {
+            switch $0 {
+            case
+                let .target(name: name, condition: _),
+                let .project(target: name, path: _, condition: _):
+                return name
+            case .library, .framework, .package, .sdk, .xcframework, .xctest:
+                return nil
+            }
+        }
+            
+        let frameworkSearchPaths = movedProductNames.map {
+            "$(CONFIGURATION_BUILD_DIR)$(TARGET_BUILD_SUBPATH)/\($0)"
+        }
+        
+        let copyProductsScript = movedProductNames
+            .map {
+                """
+                #!/usr/
+                
+                ln -s "$CONFIGURATION_BUILD_DIR/\($0)/\($0).framework"  "$CONFIGURATION_BUILD_DIR$TARGET_BUILD_SUBPATH/$PRODUCT_NAME/\($0).framework"
+                """
+            }
+            .joined(separator: "\n")
+        
+        var additionalSettings: SettingsDictionary = [
+            "FRAMEWORK_SEARCH_PATHS": .array(frameworkSearchPaths)
+        ]
+
+        additionalSettings["TARGET_BUILD_DIR"] = "$(CONFIGURATION_BUILD_DIR)$(TARGET_BUILD_SUBPATH)/$(PRODUCT_NAME)"
+
+        additionalSettings["BUILT_PRODUCTS_DIR"] = "$(CONFIGURATION_BUILD_DIR)$(TARGET_BUILD_SUBPATH)/$(PRODUCT_NAME)"
+
+        return (
+            target.with(
+                additionalSettings: additionalSettings
+            )
+            .with(
+                scripts: [
+                    TargetScript(
+                        name: "Copy Build Products",
+                        order: .pre,
+                        script: .embedded(copyProductsScript)
+                    )
```

Comment:
Isn't it possible to tell Xcode to build a product directly into a different directory? I thought that'd be controllable via build settings. If not, wouldn't we still have the same issue since we are only creating a `symbolic` link? I think I'm missing something in this logic.